tests/rptest/scale_tests/many_clients_test.py

```diff
@@ -45,9 +50,6 @@ def test_many_clients(self):
         than usual.
         """
 
-        # This test requires dedicated system resources to run reliably.
-        assert self.redpanda.dedicated_nodes
```

Comment:
Was this intentional or just for local testing?

Reply:
It's intentional -- it is for local testing, but makes sense to commit it so that next person working on the test doesn't have to hack around it.  Now that we have a clearly separate scale_tests/ folder the individual tests can avoid asserting on this kind of thing.